io.asm

```diff
@@ -278,3 +288,58 @@ RESTORE_INPUT
     jsr pop_input_stack
     sta READ_EOF
     rts
+
+; handle errors returned by open,
+; close, and chkin. If ioresult is
+; nonzero, print error message and
+; abort.
+    +BACKLINK "ioabort", 7
+IOABORT ; ( ioresult -- )
+    inx
+    lda LSB-1,x
+    ora MSB-1,x
+    bne +
+    rts
++   lda LSB-1,x
+    cmp #10
+    bcc .print_basic_error
+
+    ; prints "ioerr"
+    lda #<.ioerr
+    sta W
+    lda #>.ioerr
+    sta W+1
+    jmp .print_msb_terminated_string
+
+.ioerr
+    !text "ioer"
+    !byte 'r'|$80
+
+.print_basic_error
+    lda #$37
+    sta 1
+
+    lda LSB-1,x
+    asl
+    tax
+    lda $a326,x
+    sta W
+    lda $a327,x
+    sta W+1
+
+.print_msb_terminated_string
+    jsr CLRCHN
```

Comment:
i hope `close_all_logical_files` in `ABORT` will still work safely, even if I call CLRCHN first.

Reply:
Theoretically, it isn't that difficult to see how many files and which are open. I'll look around and maybe prototype something up.

Reply:
CLRCHN just sets screen and keyboard and sends UNTALK or UNLISTEN to appropriate devices.

Reply:
hopefully it will work just fine still.